### DISCUSSION TITLE
When openedge-project.json can't be in the source path?

### DISCUSSION THREAD

**Discussion by stage3-net** [vscode-abl/vscode-abl#140] [Q&A] [answered]
As stated, _openedge-project.json_ has to be in the root path of a folder containing a source file. But my workspaces contain paths on remote filesystems that do not allow config files, hence the extension reports "Current buffer doesn’t belong to any OpenEdge project" when editing these files.
 
Is there a workaround? It’d be great if I could define a path to _openedge-project.json_ in my workspace config and have it apply for all paths in the workspace.  Or designate a (local) workspace path as the config source to be used for the workspaceContains search.

Thanks,
Steve

**Comment by gquerret** [comment, marked as answer]
It's currently not possible to work this way. Not sure if this is your use case or not, but you can use the SSH extension to open ABL projects on remote systems.

**Comment by stage3-net** [comment]
Same issue when accessing via SSH. Most of the time source files to be edited in our projects need to be accessible to a common run-time environment for testing. On linux I can probably link the files I am editing under a personal folder that contains the config, but I don't think that will work on windows with samba mounted drives.

**Comment by stage3-net** [comment]
I tried a couple of ways to trick the extension into working without success, apparently the workspaceContains search is thrown off when a workspace contains multiple folders.

@gquerret do you have any plans to support vscode workspaces ("multi-root workspaces") with the extension? Having that feature would be valuable for corporate/enterprise projects. See https://github.com/microsoft/vscode/wiki/Adopting-Multi-Root-Workspace-APIs for more info.

**Comment by gquerret** [comment]
@stage3-net I still had a `.code-workspace` lying on my disk, so I tried it. The two projects in this file were opened correctly (even with projects on different drive letters).

**Reply by PeterJudgeZA** [reply]
FWIW I/we use code workspaces all the time. But it may not help @stage3-net since that still requires the openedge-project.json file to be in the project location.

**Reply by stage3-net** [reply]
I was able to add a windows symlink for the network path (_mklink /D c:\mnt \\\\corpdrive\src_) and it works great to allow me to store openedge-project.json in a personal folder.
Yet the ABL extension crashes when I have an ssh drive mounted with the SSH FS extension, which I have in all of my workspaces. Unfortunately I can't use the msft Remote-SSH extension because it violates company policy due to its required server-side code. SSH FS does not have this limitation.
